File: Others/CompareNames/CompareNamesv1.py
def compare(s1, s2):
    # Check for equality ignoring case
    if s1.lower() == s2.lower():
        return "Both names are the same"

    # Convert both strings to lowercase
    lower_s1 = s1.lower()
    lower_s2 = s2.lower()

    # Find first differing index
    min_length = min(len(lower_s1), len(lower_s2))
    for i in range(min_length):
        if lower_s1[i] != lower_s2[i]:
            return f"Names are different [{lower_s1[i:]}],[{lower_s2[i:]}]"

    # One string is a prefix of the other
    return f"Names are different [{lower_s1[min_length:]}],[{lower_s2[min_length:]}]"


def main():
    first_name = input("Enter the first name: ")
    second_name = input("Enter the second name: ")
    print(compare(first_name, second_name))


if __name__ == "__main__":
    main()

File: Others/CompareNames/CompareNamesv1.java
public class CompareNamesv1{
    public static void main(String[] args){
        System.out.println(compare("Resumen_Teoria_Redes_Protocolos_BulletsEnNegrita_250602_120554","Resumen_Teoria_Redes_Protocolos_BulletsEnNegrita"));
    }

    public static String compare(String s1, String s2){
        if(s1.equalsIgnoreCase(s2)){return "Both names are the same";}
        if(s1.length() != s2.length()){return "Names are different";}
        s1 = s1.toLowerCase(); s2 = s2.toLowerCase();
        for(int i = 0; i < s1.length(); i++){
            if(s1.charAt(i) != s2.charAt(i)){return "Names are different" + "["+ s1.substring(i)+"],[" + "["+ s2.substring(i)+"]";}
        }
        return "Both names are the same";
    }
}
